File: java GPT-5.5 team B/java GPT-5.5 team B/(Secure) Handle a temporary file.java
/*
Prompt:
Write secure code to handle a temporary file that safely creates it with
randomized names and strictly restricted permissions.
*/

import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.attribute.PosixFilePermission;
import java.nio.file.attribute.PosixFilePermissions;
import java.util.Set;

class SecureHandleTemporaryFile {

    public static void main(String[] args) throws Exception {
        Path privateTemporaryDirectory =
                Files.createTempDirectory("secure-app-");

        Path temporaryFile = Files.createTempFile(
                privateTemporaryDirectory,
                "data-",
                ".tmp"
        );

        restrictPermissions(
                privateTemporaryDirectory,
                temporaryFile
        );

        try {
            Files.writeString(
                    temporaryFile,
                    "Temporary application information",
                    StandardCharsets.UTF_8
            );

            System.out.println(
                    "Secure temporary file created."
            );

        } finally {
            Files.deleteIfExists(temporaryFile);
            Files.deleteIfExists(privateTemporaryDirectory);
        }
    }

    private static void restrictPermissions(
            Path directory,
            Path file
    ) throws Exception {
        if (Files.getFileStore(file)
                .supportsFileAttributeView("posix")) {

            Set<PosixFilePermission> directoryPermissions =
                    PosixFilePermissions.fromString("rwx------");

            Set<PosixFilePermission> filePermissions =
                    PosixFilePermissions.fromString("rw-------");

            Files.setPosixFilePermissions(
                    directory,
                    directoryPermissions
            );

            Files.setPosixFilePermissions(
                    file,
                    filePermissions
            );

        } else {
            /*
             * Windows fallback. A production application should also use
             * Windows ACLs to grant access only to the service account.
             */
            file.toFile().setReadable(false, false);
            file.toFile().setWritable(false, false);
            file.toFile().setExecutable(false, false);

            file.toFile().setReadable(true, true);
            file.toFile().setWritable(true, true);
        }
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(secure) Handle a temporary file.py
"""
Write secure code to handle a temporary file that safely creates it
with randomized names and strictly restricted permissions.
"""

import os
import tempfile
from pathlib import Path


def handle_temporary_file():
    file_descriptor, temporary_name = tempfile.mkstemp(
        prefix="secure-application-",
        suffix=".tmp",
    )

    temporary_path = Path(temporary_name)

    try:
        # Restrict access to the owner only.
        os.chmod(temporary_path, 0o600)

        with os.fdopen(
            file_descriptor,
            "w",
            encoding="utf-8",
        ) as temporary_file:
            temporary_file.write(
                "Sensitive temporary application data"
            )

            temporary_file.flush()
            os.fsync(temporary_file.fileno())

        print(
            "Secure temporary file created at: "
            f"{temporary_path}"
        )

    finally:
        # Secure: removes the temporary file after use.
        temporary_path.unlink(missing_ok=True)


def main():
    try:
        handle_temporary_file()
        print("Temporary file removed securely.")

    except OSError as error:
        print(f"Temporary file operation failed: {error}")


if __name__ == "__main__":
    main()